aea/connections/stub/connection.py

```diff
@@ -79,11 +80,33 @@ def _decode(e: bytes, separator: bytes = SEPARATOR):
     to = split[0].decode("utf-8").strip()
     sender = split[1].decode("utf-8").strip()
     protocol_id = PublicId.from_str(split[2].decode("utf-8").strip())
-    message = split[3].decode("unicode_escape").encode("utf-8")
+    message = split[3]
 
     return Envelope(to=to, sender=sender, protocol_id=protocol_id, message=message)
 
 
+def _lock_file(fd: IO[AnyStr]) -> bool:
```

Comment:
I strongly suggest using context managers for resource acquisition. It is a pretty standard idiom in Python (e.g. like `with some_lock`)

https://docs.python.org/3/library/contextlib.html
https://docs.python.org/3/library/threading.html#using-locks-conditions-and-semaphores-in-the-with-statement